### PR TITLE
AG -9077 (subtask: AG-9156)- First pass of interface documentation

### PR DESCRIPTION
**Pull request by taktran**
## Changes

* Add `ApiDocumentation` and `InterfaceDocumentation` components from grid
* Add `interfaceDocumentation` markdoc tag
* Update all docs to use `interfaceDocumentation` markdoc tag
* Add prism types

## Known issues

* `ApiDocumentation` component not done yet - `charts-api-create-update` and `charts-api-download`
* Code details does not recursively show all interfaces
* Defaults values are not included
* Required fields are not highlighted and not sorted to the top

### CODE REVIEW

File: packages/ag-charts-website/src/content/docs/charts-radar-series/index.mdoc [hunk 1]
```diff
@@ -63,132 +63,6 @@ axes: [
...
-    tick?: AgAxisNumberTickOptions;
-    /** Shape of grid lines. Default: `polygon` */
-    shape?: 'polygon' | 'circle';
-    /** Configuration for the title shown next to the axis. */
-    title?: AgAxisCaptionOptions;
-}
-```
+## API Reference
+
+{% interfaceDocumentation interfaceName="AgRadarAreaSeriesOptions" config="{ \"showSnippets\": false }" /%}
```
Comment (alantreadway) [inline packages/ag-charts-website/src/content/docs/charts-radar-series/index.mdoc:70]: Nit: This should be for `AgRadarLineSeriesOptions` too (there are two series types in this page at the moment)

File: packages/ag-charts-website/src/features/api-documentation/components/ApiDocumentation.astro [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+import { ApiDocumentation } from './ApiDocumentation';
+import { getJsonFromDevFile } from '@utils/getJsonFromDevFile';
+
+const interfaceLookup = getJsonFromDevFile('ag-charts-community/interfaces.AUTO.json');
```
Comment (alantreadway) [inline packages/ag-charts-website/src/features/api-documentation/components/ApiDocumentation.astro:5]: Is this a server-side or client-side load? (I'm reading this as the former?)

**Review by alantreadway — APPROVED** [review]
Nice - couple of comments but nothing serious 👍

File: packages/ag-charts-website/src/content/docs/charts-radar-series/index.mdoc (hunk 1 above)
Comment (taktran) [reply packages/ag-charts-website/src/content/docs/charts-radar-series/index.mdoc:70]: Updated

File: packages/ag-charts-website/src/features/api-documentation/components/ApiDocumentation.astro (hunk 2 above)
Comment (taktran) [reply packages/ag-charts-website/src/features/api-documentation/components/ApiDocumentation.astro:5]: Yep, it's server side

**Pull request merged**